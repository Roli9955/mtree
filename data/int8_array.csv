id;val
0;-9223372036854775808
1;-9223372036854775808,0
2;-9223372036854775808,0,9223372036854775807