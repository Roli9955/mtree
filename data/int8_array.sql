/* contrib/mtree_gist/data/int8_array.sql */

DROP TABLE IF EXISTS INT8_ARRAY_TEST CASCADE;
CREATE TABLE INT8_ARRAY_TEST (
  id  INTEGER,
  val mtree_int8_array
);
COPY INT8_ARRAY_TEST(id, val) FROM '/home/postgres/test_files/int8_array.csv' DELIMITER ';' CSV HEADER;

DROP INDEX IF EXISTS INT8_ARRAY_TEST_IDX CASCADE;
CREATE INDEX INT8_ARRAY_TEST_IDX ON INT8_ARRAY_TEST USING GiST (val mtree_int8_array_opclass);

SET enable_seqscan TO OFF;

SELECT * FROM INT8_ARRAY_TEST;

SELECT id, val, (val <-> '0') AS dst FROM INT8_ARRAY_TEST ORDER BY (val <-> '0');

SELECT * FROM INT8_ARRAY_TEST WHERE val #<# '0';
