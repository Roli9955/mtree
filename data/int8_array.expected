SET
DROP TABLE
CREATE TABLE
COPY 3
DROP INDEX
CREATE INDEX
SET
 id |                    val                     
----+--------------------------------------------
  0 | 0
  1 | -9223372036854775808,0
  2 | -9223372036854775808,0,9223372036854775807
(3 rows)
